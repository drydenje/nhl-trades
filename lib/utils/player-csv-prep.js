"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.addHrIdsToPlayerList = void 0;
var _convertArrayToCsv = require("convert-array-to-csv");
var _utils = require("./utils");
const addHrIdsToPlayerList = (players, hrPlayers) => {
  const duplicates = [];

  // the list of players that will be returned
  let finalPlayersArray = players.map(player => {
    console.log("Player:", player);
    // for each player in the nhl.com list

    // select the player from the hr list
    const hrFindPlayer = hrPlayers.filter(hrPlayer => {
      const comparison = hrPlayer.name.localeCompare(player.name, "en-US", {
        sensitivity: "base"
      });
      return comparison ? false : true;
    });

    // console.log("FIND:", hrFindPlayer);
    if (hrFindPlayer.length > 1) {
      duplicates.push(player);
      return {
        ...player
      };
    } else {
      return {
        ...player,
        ...(hrFindPlayer[0] ? {
          hrID: hrFindPlayer[0].hrID
        } : null)
      };
    }
  });
  (0, _utils.writeFile)(`./public/scraped-data/duplicate-ids.json`, duplicates);
  console.log("Duplicates:", duplicates);
  return finalPlayersArray;
};
exports.addHrIdsToPlayerList = addHrIdsToPlayerList;