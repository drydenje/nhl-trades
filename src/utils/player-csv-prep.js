import { convertArrayToCSV } from "convert-array-to-csv";

const addHrIdsToPlayerList = (players, hrPlayers) => {
  // the list of players that will be returned
  let finalPlayersArray = players.map((player) => {
    // for each player in the nhl.com list

    // select the player from the hr list
    const hrFindPlayer = hrPlayers.filter((hrPlayer) => {
      // console.log("hr + nhl:", hrPlayer.name, player.name);
      const a = hrPlayer.name.localeCompare(player.name, "en-US", {
        sensitivity: "base",
      });
      // console.log("a:", a);
      return a;
    });

    return {
      ...player,
      ...(hrFindPlayer[0] === 0 ? { hrID: hrFindPlayer[0].hrID } : null),
    };
  });

  return finalPlayersArray;
};

export { addHrIdsToPlayerList };

//  const expectedList = [
//    {
//      id: 8444924,
//      name: "Jim Anderson",
//      hrID: "anderji01",
//    },
//    {
//      id: 8444926,
//      name: "Murray Anderson",
//      hrID: "andermu01",
//    },
//    {
//      id: 8444995,
//      name: "Pete Backor",
//      hrID: "backope01",
//    },
//  ];
