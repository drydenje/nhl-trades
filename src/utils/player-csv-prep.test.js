import { addHrIdsToPlayerList } from "./player-csv-prep.js";
// import { server, http, HttpResponse } from "../config/testServer.js";

// const DATA_FILEPATH = `./public/scraped-data/`;

// describe("add-nhl-players function", () => {
//   // this shouldn't return csv?
//   test("should return a csv (file?) given an array of nhl.com players", () => {
//     const headings = ["id", "name", "hrId"];
//     const players = [
//       {
//         id: 8444924,
//         name: "Jim Anderson",
//       },
//       {
//         id: 8444926,
//         name: "Murray Anderson",
//       },
//       {
//         id: 8444995,
//         name: "Pete Backor",
//       },
//     ];

//     // const yearToCheck = "2021-22";
//     // const result = checkArray(seasons[yearToCheck]);
//     // expect(result).toEqual(1);
//   });
// });

describe("add-hr-players function", () => {
  test("should add hr id's to the player array", () => {
    const players = [
      {
        id: 8444924,
        name: "Jim Anderson",
      },
      {
        id: 8444926,
        name: "Murray Anderson",
      },
      {
        id: 8444995,
        name: "Pete Backor",
      },
    ];

    const hrPlayers = [
      { name: "Jim Anderson", hrID: "anderji01" },
      { name: "Murray Anderson", hrID: "andermu01" },
      { name: "Pete Backor", hrID: "backope01" },
    ];

    const expectedList = [
      {
        id: 8444924,
        name: "Jim Anderson",
        hrID: "anderji01",
      },
      {
        id: 8444926,
        name: "Murray Anderson",
        hrID: "andermu01",
      },
      {
        id: 8444995,
        name: "Pete Backor",
        hrID: "backope01",
      },
    ];

    const result = addHrIdsToPlayerList(players, hrPlayers);
    expect(result.length).toEqual(3);
    expect(result).toMatchSnapshot();
  });

  test("should add hockey-reference.com id's to the player array when the player name contains special characters", () => {
    const players = [
      {
        id: 8460712,
        name: "Tomas Vokoun",
      },
      {
        id: 8469689,
        name: "Tomas Mojzis",
      },
      {
        id: 8478007,
        name: "Elvis Merzlikins",
      },
    ];

    const hrPlayers = [
      { name: "Elvis Merzļikins", hrID: "merzlel01" },
      { name: "Tomáš Vokoun", hrID: "vokouto01" },
      { name: "Tomáš Mojžíš", hrID: "mojzito01" },
    ];

    const expectedList = [
      {
        id: 8460712,
        name: "Tomas Vokoun",
        hrID: "vokouto01",
      },
      {
        id: 8469689,
        name: "Tomas Mojzis",
        hrID: "mojzito01",
      },
      {
        id: 8478007,
        name: "Elvis Merzlikins",
        hrID: "merzlel01",
      },
    ];

    const result = addHrIdsToPlayerList(players, hrPlayers);
    expect(result.length).toEqual(3);
    expect(result).toMatchSnapshot();
  });
});
