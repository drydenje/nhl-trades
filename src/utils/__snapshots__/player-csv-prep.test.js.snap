// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-hr-players function should add hockey-reference.com id's to the player array when the player name contains special characters 1`] = `
[
  {
    "id": 8460712,
    "name": "Tomas Vokoun",
    "hrID": "vokouto01",
  },
  {
    "id": 8469689,
    "name": "Tomas Mojzis",
    "hrID": "mojzito01",
  },
  {
    "id": 8478007,
    "name": "Elvis Merzlikins",
    "hrID": "merzlel01",
  },
]
`;

exports[`add-hr-players function should add hr id's to the player array 1`] = `
[
  {
    "id": 8444924,
    "name": "Jim Anderson",
  },
  {
    "id": 8444926,
    "name": "Murray Anderson",
  },
  {
    "id": 8444995,
    "name": "Pete Backor",
  },
]
`;
